mlir/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -1653,6 +1653,26 @@ void mlir::scalarReplace(AffineForOp forOp) {
   LLVM_DEBUG(llvm::dbgs() << "SCAL REP END\n");
 }
 
+void mlir::affineParallelize(AffineForOp forOp) {
+  OpBuilder outsideBuilder(forOp);
+  Location loc = forOp.getLoc();
+  BlockAndValueMapping operandMap;
+  auto forOpIV = forOp.getInductionVar();
+  auto newPloop = outsideBuilder.create<AffineParallelOp>(
+      loc, forOp.getLowerBoundMap(), forOp.getLowerBoundOperands(),
+      forOp.getUpperBoundMap(), forOp.getUpperBoundOperands());
+  auto parallelIV = newPloop.getIVs()[0];
+  operandMap.map(forOpIV, parallelIV);
+  OpBuilder ob(newPloop);
+  ob.setInsertionPoint(newPloop.getBody(), newPloop.getBody()->begin());
+  Block::iterator srcBlockEnd = forOp.getBody()->end();
+  for (auto it = forOp.getBody()->begin(); it != std::prev(srcBlockEnd); it++) {
+    ob.clone(*it, operandMap);
+  }
+  forOp.erase();
+  llvm::dbgs() << "PAR CONV END\n";
```

Comment:
You can't leave debug output unguarded - this can be removed; not useful for users.